x-pack/plugins/elastic_assistant/server/routes/knowledge_base/get_knowledge_base_status.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IRouter } from '@kbn/core/server';
+import { transformError } from '@kbn/securitysolution-es-utils';
+
+import { buildResponse } from '../../lib/build_response';
+import { buildRouteValidation } from '../../schemas/common';
+import { ElasticAssistantRequestHandlerContext } from '../../types';
+import { KNOWLEDGE_BASE } from '../../../common/constants';
+import { GetKnowledgeBaseStatusPathParams } from '../../schemas/knowledge_base/get_knowledge_base_status';
+import { ElasticsearchStore } from '../../lib/langchain/elasticsearch_store/elasticsearch_store';
+import {
+  ELSER_MODEL_ID,
+  ESQL_DOCS_LOADED_QUERY,
+  ESQL_RESOURCE,
+  KNOWLEDGE_BASE_INDEX_PATTERN,
+} from './constants';
+
+export interface GetKnowledgeBaseStatusResponse {
+  elser_exists: boolean;
+  esql_exists?: boolean;
+  index_exists: boolean;
+  pipeline_exists: boolean;
+}
+
+/**
+ * Get the status of the Knowledge Base index, pipeline, and resources (collection of documents)
+ *
+ * @param router IRouter for registering routes
+ */
+export const getKnowledgeBaseStatusRoute = (
+  router: IRouter<ElasticAssistantRequestHandlerContext>
+) => {
+  router.get(
+    {
+      path: KNOWLEDGE_BASE,
+      validate: {
+        params: buildRouteValidation(GetKnowledgeBaseStatusPathParams),
+      },
+      options: {
+        // Note: Relying on current user privileges to scope an esClient.
+        // Add `access:kbnElasticAssistant` to limit API access to only users with assistant privileges
+        tags: [],
+      },
+    },
+    async (context, request, response) => {
+      const resp = buildResponse(response);
+      const logger = (await context.elasticAssistant).logger;
+
+      try {
+        const kbResource =
+          request.params.resource != null ? decodeURIComponent(request.params.resource) : undefined;
+
+        // Get a scoped esClient for finding the status of the Knowledge Base index, pipeline, and documents
+        const esClient = (await context.core).elasticsearch.client.asCurrentUser;
+        const esStore = new ElasticsearchStore(esClient, KNOWLEDGE_BASE_INDEX_PATTERN, logger);
+
+        const indexExists = await esStore.indexExists();
+        const pipelineExists = await esStore.pipelineExists();
+        const modelExists = await esStore.isModelInstalled(ELSER_MODEL_ID);
+        const esqlExists =
+          indexExists && (await esStore.similaritySearch(ESQL_DOCS_LOADED_QUERY)).length > 0;
+
+        const body: GetKnowledgeBaseStatusResponse = {
+          elser_exists: modelExists,
+          esql_exists: kbResource === ESQL_RESOURCE ? esqlExists : undefined,
```

Comment:
per our offfline chat, hapi will exclude `esql_exists` from the response when this value is `undefined`